src/main/java/Rizz.java

```diff
@@ -0,0 +1,252 @@
+                Files.createDirectories(dataFilePath.getParent());
+                Files.createFile(dataFilePath);
+            }
+
+            StringBuilder builder = new StringBuilder();
+            if (data.length > 0) {
+                builder.append(data[0]);
+                for (int i = 1; i < data.length; i++) {
+                    builder.append("\n").append(data[i]);
+                }
+            }
+            Files.writeString(dataFilePath, builder.toString());
+
+        } catch (IOException e) {
+            System.out.println("There was an error saving the tasks.");
+        }
+    }
+
+    private void loadTasks() {
+        if (Files.exists(dataFilePath)) {
+            try {
+                List<String> lines = Files.readAllLines(dataFilePath);
+                for (int i = 0; i < lines.size(); i++) {
+                    String line = lines.get(i);
+                    Task task = decodeTasks(line);
+                    if (task != null) {
+                        arrList.add(task);
+                    }
+                }
+                if (arrList.isEmpty()) {
+                    System.out.println("No saved tasks found. Starting with an empty list.");
+                } else {
+                    System.out.println("Tasks loaded successfully!");
+                }
+            } catch (IOException e) {
+                System.out.println("There was an error loading the tasks.");
+            }
+        } else {
+            System.out.println("No saved tasks found. Starting with an empty list.");
+        }
+    }
+
+    private Task decodeTasks(String line) {
+        String[] parts = line.split(" \\| ");
+        String taskType = parts[0];
+        boolean isDone = parts[1].equals("1");
+        String text = parts[2];
+
+        switch (taskType) {
+            case "T":
+                return new ToDo(text, isDone);
+            case "D":
+                try {
+                    LocalDateTime time = LocalDateTime.parse(parts[3], formatter);
+                    return new Deadline(text, time, isDone);
+                } catch (DateTimeParseException e) {
+                    System.out.println("\t The date/time format is incorrect. Please use yyyy-MM-dd HHmm.");
+                    return null;
+                }
+            case "E":
+                String from = parts[3];
+                String to = parts[4];
+                return new Event(text, from, to, isDone);
+            default:
+                System.out.println("Unknown task type: " + taskType);
+                return null;
+        }
+    }
+
+
+
+
+    private void greet() {
+        this.loadTasks();
+        String str = "\t____________________________________________________________\n" +
+                "\tHello! I'm Rizz\n" +
+                "\tWhat can I do for you?\n" +
+                "\t____________________________________________________________\n";
+        System.out.println(str);
+    }
+
+    private void exit() {
```

Comment:
Maybe this could be named as `getExitString` instead because `exit` implies that the function terminates the programme.